mtags/src/main/scala-2/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -109,57 +130,111 @@ object MultilineStringFormattingProvider {
-          var endIndex = tokenIndex + 1
-          while (!tokens(endIndex)
-              .isInstanceOf[Interpolation.End]) endIndex += 1
-          val end = tokens(endIndex)
-          stringFound = end.end > startPos.end
-          shouldAddPipes =
-            stringFound && isMultilineString(sourceText, start) &&
-              hasStripMarginSuffix(endIndex, tokens)
-        case _ =>
-      }
-      tokenIndex += 1
+    if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
+      tokens.zipWithIndex
+        .exists(
+          shouldFormatMultiString(sourceText, tokens, startPos, endPos) orElse
+            shouldFormatInterpolationString(sourceText, tokens, startPos) orElse {
+            case _ => false
+          }
+        )
+    } else false
+  }
+
+  private def shouldFormatMultiString(
+      sourceText: String,
+      tokens: Tokens,
+      start: StartPosition,
+      end: EndPosition
+  ): PartialFunction[(Token, Int), Boolean] = {
+    case (token: Constant.String, index: Int) =>
+      isMultilineString(sourceText, token) && hasStripMarginSuffix(
+        index,
+        tokens
+      ) && inToken(start, end, token)
+  }
+  private def shouldFormatInterpolationString(
+      sourceText: String,
+      tokens: Tokens,
+      start: StartPosition
+  ): PartialFunction[(Token, Int), Boolean] = {
+    case (token: Interpolation.Start, index: Int)
+        if token.start < start.start => {
+      var endIndex = index + 1
+      while (!tokens(endIndex)
+          .isInstanceOf[Interpolation.End]) endIndex += 1
+      isMultilineString(sourceText, token) && hasStripMarginSuffix(
+        endIndex,
+        tokens
+      )
     }
-    shouldAddPipes
   }
 
   private def withToken(
       textId: TextDocumentIdentifier,
       sourceText: String,
-      range: Range,
-      newlineAdded: Boolean
-  )(fn: (String, meta.Position) => List[TextEdit]): List[TextEdit] = {
+      range: Range
+  )(
+      fn: (
+          StartPosition,
+          EndPosition,
+          String,
+          Option[Tokens]
+      ) => List[TextEdit]
+  ): List[TextEdit] = {
     val source = textId.getUri.toAbsolutePath
     if (source.exists) {
       val virtualFile = Input.VirtualFile(source.toString(), sourceText)
       val startPos = range.getStart.toMeta(virtualFile)
       val endPos = range.getEnd.toMeta(virtualFile)
-      if (pipeInScope(startPos, endPos, sourceText, newlineAdded)) {
-        val tokens =
-          virtualFile.tokenize.toOption
-        tokens
-          .filter(multilineStringInTokens(_, startPos, endPos, sourceText))
-          .map(_ => fn(sourceText, startPos))
-          .getOrElse(Nil)
-      } else Nil
+      fn(startPos, endPos, sourceText, virtualFile.tokenize.toOption)
     } else Nil
   }
 
+  private def doubleQuoteNotClosed(
+      splitLines: Array[String],
+      position: Position
+  ): Boolean = {
+    val lineBefore = splitLines(position.getLine - 1)
+    !getIndexOfLastQuote(lineBefore).exists(_._2)
```

Comment:
MAybe we can make it a bit more clear:
`getIndexOfLastQuote(lineBefore).exists{ case (_, quoteClosed) => !quoteClosed }`

Reply:
ok